### ISSUE TITLE
Fine-tune USE on domain specific corpus

### ISSUE THREAD

**Issue by AmmarRashed** [tensorflow/hub#597]
Hi,
Could you please shed some light on how to fine-tune pre-trained USE models on domain specific corpus? Vectors just need to be fine-tuned to a domain-specific context. There is no downstream supervised task or anything.

This here used to work for TF1, but TF2 deprecated hub.Module
https://github.com/helloeve/universal-sentence-encoder-fine-tune

Thanks,

**Comment by djdongjin** [comment]
Hi, #600 might be helpful. Also, the fine-tuning in your link seems supervised fine-tuning, since the sentences are classified. If that's your usecase, you can construct a dataset similarly, and concatenate the `hub.KerasLayer(...)` with a dense layer and train with cross-entropy loss, as shown in the first comment of #600. If you don't have the similarity info of sentences, the latter comments might be more relevant.

**Comment by gowthamkpr** [comment]
@AmmarRashed You should be able to use USE on domain specific corpus. If the domain has lots of vocab that is not commonly observed in typical language data, an adaption training would be better suited. 
As mentioned in the above comment ass a dense layer on top of `hub.KerasLayer()` and it should solve your issue.

**Comment by AmmarRashed** [comment]
Thanks. That was helpful. I was hoping for a more straightforward solution like:
`model = hub.load("https://tfhub.dev/google/universal-sentence-encoder-multilingual/3", trainable=True)
model.finetune([ "list of sentences"])`
Although the model is really good, in production the usage keeps changing, and the embedding scores need to adapt to the style used by the users.

**Comment by anirbankonar123** [comment]
As given here, i am also looking for a way to fine-tune the embeddings for my own domain specific corpus. Is there a way currently ? I dnt have labels, and cannot do Supervised training. Just need to fine-tune the embeddings, as given in the post above. Please update me. Thanks.

**Comment by gowthamkpr** [comment]
@anirbankonar123 This discussion is going on in this issue [here](https://github.com/tensorflow/hub/issues/600). Please take a look at that issue. Thanks!

**Comment by AmmarRashed** [comment]
Thanks. Closing this issue as it is already being discussed in #600
